boruta/script.whizzml

```diff
@@ -168,24 +212,65 @@
 ;;   fields: (map) Fields information structure
 ;;   f-names: (list) Fields names
 ;;   objective: (string) objective field name
+;;   median-shadows: (number) Median value of the list of maximum shadow importances of all executions
+;;   f-importanes: (list) Field importances from all executions
 ;;
 ;; Output:  (map) Map from categories (important, unimportant or undecided) to
 ;;                list of field names
 ;;
-(define (group-importance f-rank fields f-names objective)
+(define (group-importance f-rank fields f-names objective, median-shadows, f-importances)
   (let (to-name (lambda (x) (fields [x "name"] false))
         important (iterate (r [] k (keys f-rank))
                     (if (positive? (f-rank k)) (append r (to-name k)) r))
-        important (append important objective)
+        ;; Objective is also an important field
+        important (append important objective) 
         undecided (iterate (r [] k (keys f-rank))
                     (if (zero? (f-rank k)) (append r (to-name k)) r))
         unimportant (filter (lambda (x)
                               (not (or (member? x important)
                                        (member? x undecided))))
-                            f-names))
+                            f-names)
+        ;; Obtain id of undecided fields
+        undecided-ids (iterate (r [] k (keys f-rank))
+                    (if (zero? (f-rank k)) (append r k) r))
+        ;; Calculating if undecided fields are important or unimportant
+        undecided-unimportant (undecided-unimportance undecided-ids median-shadows f-importances fields)
+        undecided-important  (filter (lambda (x) (not  (member? x undecided-unimportant) )) undecided)
+        
+        _ (log-info " ---- Final results: ----- ")
+        _ (log-info " Important: "important)
+        _ (log-info " Unimportant" unimportant)
+        _ (log-info " Undecided-important" undecided-important)
+        _ (log-info " Undecided-unimportant" undecided-unimportant)
+
+        ;; Add undecided-important and undecided-unimportant fields to
+        ;; important and unimportant lists
+        important (concat important undecided-important)
+        unimportant (concat unimportant undecided-unimportant)
+
+        ; Remove created resources
+        _ (for (id (created-resources)) (safe-delete id))
```

Comment:
all the `_ log` lines can be put in the body of the let

(let (...)
  (log-info "foo")
  (log-info "bar")
  (map safe-delete (created-resources))
  {"important" ...
```